qml/DomainGraphLayout.qml

```diff
@@ -345,15 +361,28 @@ Item
                             ,"y": endpoint_topic_connections_[key]["y"] - (connection_thickness_ / 2)
                             ,"width": topic_locations_[topic_id]["width"]
                             ,"height":connection_thickness_, "z":200, "left_margin": 2*elements_spacing_
-                            ,"arrow_color": topic_color_, "background_color": background_color.color }
+                            ,"arrow_color": topic_color_, "background_color": background_color.color
+                            ,"endpoint_id": key }
                         var connection_bar = arrow_component.createObject(topic_connections, input)
                         topic_painted_[topic_painted_.length] = key;
                     }
                 }
             }
+
+            topic_connections_generated = true
```

Comment:
How many times is this variable set to true? Once every topic has generated its connections? Same for endpoints.
In that case, it would tell graph has been initialized after the first topic and endpoint have their connections generated.

Reply:
Once per loaded model. The topic connections are generated (all) at once. Same as endpoints.